core/src/main/java/com/exadel/etoolbox/linkinspector/core/services/ext/CustomLinkResolverImpl.java

```diff
@@ -0,0 +1,40 @@
+package com.exadel.etoolbox.linkinspector.core.services.ext;
+
+import com.exadel.etoolbox.linkinspector.api.service.LinkTypeProvider;
+import com.exadel.etoolbox.linkinspector.core.models.Link;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Component(service = CustomLinkResolver.class)
+public class CustomLinkResolverImpl implements CustomLinkResolver{
+    private List<LinkTypeProvider> providers = new ArrayList<>();
+
+    @Override
+    public List<LinkTypeProvider> getLinkTypeProviders() {
+        return providers;
+    }
+
+    @Override
+    public List<Link> getLinks(String propertyValue){
+        return providers.stream().filter(p->p.getLinkValue(propertyValue)!=null)
```

Comment:
Extracting the link value can e non-trivial. E.g., when one begs to a RegExp. In such a case, it is better to avoid retrieving a link value twice in a row. Therefore,
```
return providers
  .stream()
  .map(p -> tryGetLink(p, propertyValue)
  .filter(Objects::notNull)
  .collect(Collectors.toList());

private static Link tryGetLink(Provider provider, String value) {
  String linkValue = provider.getLinkValue(value);
  return StringUtils.isNotBlank(linkValue)
    ? new Link(provider, linkValue)
    : null;
}
```